blackjax/sgmcmc/diffusion.py

```diff
@@ -35,6 +35,102 @@ def one_step(
 
     return one_step
 
+""" 
+    We are interested in the simulations of exp(-U(x) / T), 
+    where U is energy function and T is the temperature.
+
+    logprob normally means U(x)/ T instead of U(x).
+"""
+
+
+class ContourState(NamedTuple):
+    position: PyTree
+    logprob_grad: PyTree
+    energy_pdf: PyTree
+    energy_idx: int
+
+
+def logposterior_estimator_fn(position: PyTree, data_batch: PyTree) -> float:
+    """Returns an approximation of the log-posterior density, 
+       which is often required by Advanved Markov chain Monte Carlo algorithms.
+       I am re-using the logposterior_estimator_fn in file ./sgmcmc/diffusion.py
+    Parameters
+    ----------
+    position
+        The current value of the random variables.
+    batch
+        The current batch of data
+    Returns
+    -------
+    An approximation of the value of the log-posterior density function for
+    the current value of the random variables.
+    """
+    logprior = logprior_fn(position)
+    batch_loglikelihood = jax.vmap(loglikelihood_fn, in_axes=(None, 0))
+    return logprior + data_size * jnp.mean(
+        batch_loglikelihood(position, data_batch), axis=0
+    )
+
+
+def csgld(logprob_grad_fn, 
+          zeta: float=1, 
+          temperature: float=0.01, 
+          partition: int=512,
+          energy_gap: float=100
+          total_samples: float=60000):
```

Comment:
`total_samples` is not used in the function

Reply:
Would it be possible to describe the parameters in the docstring?

Reply:
No problem. The reason of using total_samples is that in maximum likelihood, the loss is summed instead averaged. If we want to apply a sampler to data-driven problems. We may need to decide if we want to change the whole averaged loss to summed loss (that is why I am using total_samples).